internal/charmstore/search.go

```diff
@@ -112,6 +112,18 @@ func (s *Store) UpdateSearch(r *router.ResolvedURL) error {
 	return nil
 }
 
+// DeleteFromSearch remove the search record for the entity reference r.
+func (s *Store) DeleteFromSearch(entity *mongodoc.Entity) error {
```

Comment:
Why not just call s.ES.delete from Store.DeleteEntity ?
I'm not sure we need to have another exported method here.